orbit/cmd/orbit/orbit.go

```diff
@@ -616,6 +617,38 @@ func main() {
 			Platform:       osqueryHostInfo.Platform,
 		}
 
+		// Get the hardware UUID. We use a temporary osquery DB location in order to guarantee that
+		// we're getting true UUID, not a cached UUID. See
+		// https://github.com/fleetdm/fleet/issues/17934 and
+		// https://github.com/osquery/osquery/issues/7509 for more details.
+
+		tmpDBPath := filepath.Join(os.TempDir(), strings.Join([]string{uuid.NewString(), "tmp-db"}, "-"))
+		oi, err := getHostInfo(osquerydPath, tmpDBPath)
+		if err != nil {
+			return fmt.Errorf("get UUID from temp db: %w", err)
+		}
+
+		defer func() {
+			if tmpErr := os.RemoveAll(tmpDBPath); tmpErr != nil {
```

Comment:
@lucasmrod is this a good way of handling the error in the defer? I didn't see a similar situation in the codebase to go off of. This is how I've seen it done before, but this is also in the `main` function, so not sure if it will behave as expected.